xlators/features/shard/src/shard.c

```diff
@@ -548,19 +548,21 @@ shard_local_wipe(shard_local_t *local)
 }
 
 int
-shard_modify_size_and_block_count(struct iatt *stbuf, dict_t *dict)
+shard_modify_size_and_block_count(struct iatt *stbuf, dict_t *dict,
+                                  gf_boolean_t logerror)
 {
     int ret = -1;
     void *size_attr = NULL;
     uint64_t size_array[4];
 
     ret = dict_get_ptr(dict, GF_XATTR_SHARD_FILE_SIZE, &size_attr);
     if (ret) {
-        gf_msg_callingfn(THIS->name, GF_LOG_ERROR, 0,
-                         SHARD_MSG_INTERNAL_XATTR_MISSING,
-                         "Failed to "
-                         "get " GF_XATTR_SHARD_FILE_SIZE " for %s",
-                         uuid_utoa(stbuf->ia_gfid));
+        if (logerror)
+            gf_msg_callingfn(THIS->name, GF_LOG_ERROR, 0,
+                             SHARD_MSG_INTERNAL_XATTR_MISSING,
+                             "Failed to "
```

Comment:
nit could be in one line, for easier grep

Reply:
Done